zio/csupio/vectorreader.go

```diff
@@ -22,59 +21,60 @@ type VectorReader struct {
 
 	activeReaders *atomic.Int64
 	stream        *stream
-	metaFilters   *sync.Pool
+	pushdown      zbuf.Pushdown
+	metaFilter    *metafilter
 	projection    field.Projection
```

Comment:
Nit: Drop projection since it's now available as pushdown.Projection().

```suggestion
	pushdown      zbuf.Pushdown
	metaFilter    *metafilter
```

Reply:
From compiler/kernel/pushdown.go, it looks like the projection and metaProjection are two different things or am I missing something?

Reply:
I think you might be. VectorReader.project is initialized to pushdown. VectorReader.projection is initialized to pushdown.Projection(). So the projection field isn't necessary and tests pass with the following diff:

Reply:
gotcha